compiler/qsc_frontend/src/typeck/infer.rs

```diff
@@ -590,7 +590,7 @@ impl Solver {
             (&Ty::Infer(infer), ty) | (ty, &Ty::Infer(infer)) if !contains_infer_ty(infer, ty) => {
                 self.bind_ty(infer, ty.clone(), span)
             }
-            (Ty::Param(name1), Ty::Param(name2)) if name1 == name2 => Vec::new(),
+            (Ty::Param(_, name1), Ty::Param(_, name2)) if name1 == name2 => Vec::new(), // ToDo: do we need to test if the item ids are the same?
```

Comment:
Just wanted to double-check on this. It seems like a risk if the type-checker doesn't compare the item ids as well as the param ids. It seems like there'd be a risk it would equate two different type params from two different callables. Is this a case we need to cover?

Reply:
I don't think so. Remember, you've added the item ids solely for display purposes, so they aren't functional from the perspective of the compiler. A mismatch could indicate a bug in the logic that figures out what item to link for display, but the param id is fully sufficient for describing the item for type inferences purposes.

Reply:
removed todo